lib/clockwise_import.rb

```diff
@@ -0,0 +1,118 @@
+require 'csv'
+
+class ClockwiseImporter
+  def initialize(csv_path:, tenant:)
+    @csv_path = csv_path
+    @tenant = tenant
+  end
+
+  def import!
+    Apartment::Tenant.switch @tenant
+    Entry.transaction do
+      CSV.foreach(@csv_path,
+                  headers: true,
+                  quote_char: '"',
+                  col_sep: ';') do |r|
+        row = ImportRow.new(r)
+        EntryBuilder.new(row).create! unless row.hours < 1
+      end
+    end
+  end
+end
+
+class EntryBuilder
+  def initialize(row)
+    @row = row
+  end
+
+  def create!
+    client = build_client
+    project = build_project(client)
+    category = build_category
+    user = build_user
+    Entry.create!(
+      project: project,
+      category: category,
+      user: user,
+      hours: @row.hours,
+      date: @row.date,
+      description: @row.description,
+    )
+  end
+
+  private
+
+  def build_client
+    Client.where('lower(name) = ?', @row.client)
+      .first_or_create(name: @row.client)
+  end
+
+  def build_project(client)
+    Project.where('lower(name) = ? or slug = ?',
+                  @row.project.downcase,
+                  @row.project.parameterize)
+      .first_or_create(name: @row.project,
+                       billable: true,
+                       client: client,
+                       archived: @row.archive)
+  end
+
+  def build_category
+    Category.where('lower(name) = ?', @row.category.downcase)
+      .first_or_create(name: @row.category)
+  end
+
+  def build_user
+    names = @row.user.split(' ')
+    first_name = names.first
+    last_name = names.last
+    User.where('lower(first_name) = ? and lower(last_name) = ?',
+                first_name.downcase, last_name.downcase)
+                .first_or_create(first_name: first_name,
```

Comment:
Place the . on the previous line, together with the method call receiver.<br>Use 2 (not 12) spaces for indenting an expression spanning multiple lines.